dbt/adapters/dremio/credentials.py

```diff
@@ -20,34 +20,40 @@
 
 @dataclass
 class DremioCredentials(Credentials):
-    environment: Optional[str]
-    database: Optional[str]
-    schema: Optional[str]
-    datalake: Optional[str]
-    root_path: Optional[str]
+    environment: Optional[str] = None
+    UID: Optional[str] = None
+    PWD: Optional[str] = None
+    pat: Optional[str] = None
+    object_storage_source: Optional[str] = None
+    datalake: Optional[str] = None
+    object_storage_path: Optional[str] = None
+    root_path: Optional[str] = None
+    dremio_space: Optional[str] = None
+    database: Optional[str] = None
+    dremio_space_folder: Optional[str] = None
+    schema: Optional[str] = None
     cloud_project_id: Optional[str] = None
     cloud_host: Optional[str] = None
     software_host: Optional[str] = None
-    UID: Optional[str] = None
-    PWD: Optional[str] = None
     port: Optional[int] = 9047  # for rest endpoint
     use_ssl: Optional[bool] = True
-    pat: Optional[str] = None
     additional_parameters: Optional[str] = None
 
     _ALIASES = {
-        "user": "UID",
+        # Only terms on right-side will be used going forward.
         "username": "UID",
-        "pass": "PWD",
+        "user": "UID",  # backwards compatibility with existing profiles
         "password": "PWD",
-        "server": "host",
-        "track": "environment",
-        "space": "database",
-        "folder": "schema",
-        "materialization_database": "datalake",
-        "materialization_schema": "root_path",
+        "object_storage_source": "datalake",
+        "object_storage_path": "root_path",
+        "dremio_space": "database",
+        "dremio_space_folder": "schema",
+        "folder": "schema",  # backwards compatibility with existing profiles
```

Comment:
"folder" is no longer a config option, so we can remove that.

Reply:
Agreed.